client/src/components/mapping/FhirMappingPanel/FhirResourceTree/index.tsx

```diff
@@ -260,23 +251,37 @@ const FhirResourceTree = ({ onClickCallback }: Props) => {
     } else {
       if (selectedNode) selectedNode.isSelected = false;
       node.isSelected = true;
-      setSelectedNode(node);
+      setSelectedNode(node as TreeNode);
       onClickCallback(node.nodeData);
     }
     setNodes([...nodes]);
   };
 
+  const handleContextMenu = async (
+    node: ITreeNode<Attribute>
+  ): Promise<void> => {
```

Comment:
Isn't this called when you right-click on a Node? I don't get it

Reply:
yes it is, why ?

Reply:
one is for left-click, the other for right-click

Reply:
It looks like you're doing what should be done on a left-click here...

Reply:
no: the left click should fetch the children and expand the node, the right click should just fetch the nodes without collapsing the node.

Reply:
Oh, ok, on left click you generate the sub-tree inside the node so that you can tell which extensions can be applied?